addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -35,13 +49,20 @@ options.registry.TableOfContent = options.Class.extend({
      * @override
      */
     onClone: function () {
+        this._resetIds();
         this._generateNav();
     },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
 
+    /**
+     * @private
+     */
+    _resetIds() {
+        delete this.$target[0].dataset.tocId;
+    },
```

Comment:
In the end, not a fan adding an extra `data-toc-id` in the DOM. This is redundant information in the end: can't you always retrieve it by parsing the first heading id ? :+1:

Reply:
Also tested with pre-fix existing ToC in page and saved snippets.